Dockerfile

```diff
@@ -1,40 +1,93 @@
-#Nebra Helium Hotspot - BTLE Configuration Software Container
-#(C) Nebra LTD. 2021
-#Licensed under the MIT License.
+# Nebra Helium Hotspot - BTLE Configuration Software Container
+# (C) Nebra LTD. 2021
+# Licensed under the MIT License.
 
-FROM balenalib/raspberry-pi-debian:buster-run
+ARG SYSTEM_TIMEZONE="Europe/London"
 
+####################################################################################################
+################################## Stage: builder ##################################################
+
+# The balenalib/raspberry-pi-debian-python image was tested but missed many dependencies.
+FROM balenalib/raspberry-pi-debian:buster-build-20210705 as builder
+
+# Nebra uses /opt by convention
 WORKDIR /opt/
 
+# Copy python dependencies for `pip install` later
 COPY requirements.txt requirements.txt
 
+# This will be the path that venv uses for installation below
+ENV PATH="/opt/venv/bin:$PATH"
+
+# Install python3-minimal, pip3, wget, venv.
+# Then set venv environment copied from builder.
+# Finally, use pip to install dependencies.
 RUN \
     apt-get update && \
     DEBIAN_FRONTEND="noninteractive" \
-    TZ="Europe/London" \
+    TZ="$SYSTEM_TIMEZONE" \
```

Comment:
I'm unclear the significance of this value. Are there other codebases that MUST also have this value?

Reply:
I haven't run into a situation where this is required but let's leave it in for now.